service/history/timer_queue_active_task_executor.go

```diff
@@ -448,8 +452,9 @@ func (t *timerQueueActiveTaskExecutor) executeActivityRetryTimerTask(
 				tag.FailoverVersion(activityInfo.Version),
 				tag.TimerTaskStatus(activityInfo.TimerTaskStatus),
 				tag.ScheduleAttempt(task.Attempt))
+			return consts.ErrDuplicate
 		}
-		return nil
+		return consts.ErrActivityTaskNotFound
```

Comment:
I think always returning `ErrActivityTaskNotFound` should be good enough. I actually don't know why we need the `if ok` check for the logging msg.